vars/vars.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package vars includes storing and obtaining provider wide variables and their values
+package vars
+
+import (
+	"fmt"
+	"os"
+)
+
+type EnvVar struct {
+	envVarName   string
+	defaultValue string
+	isRequired   bool
+}
+
+func (ev EnvVar) GetValue() (string, error) {
+	osEnv := ev.envVarName
+	osEnvValue, isPresent := os.LookupEnv(osEnv)
+
+	if isPresent {
+		return osEnvValue, nil
+	}
+
+	if ev.isRequired {
+		return "", fmt.Errorf("%s: a required OS environment is not present", osEnv)
+	}
+
+	// Return default endpoint
+	return ev.defaultValue, nil
+}
+
+var IdentityBindingTokenEndPoint = EnvVar{
+	envVarName:   "GAIA_TOKEN_EXCHANGE_ENDPOINT",
```

Comment:
Generally we've seen trend in following with EnvVars while configuring endpoints, such as `export CLOUDSDK_API_ENDPOINT_OVERRIDES_SECRETMANAGER=SOME.ENDPOINT.googleapis.com`, This way we can point to a clean and versioned config map for injecting the env vars, so that we have a clear and explicit lifecycle for the config and for the provider pods.

Reply:
gotcha Thanks!